src/Core/Picking.js

```diff
@@ -148,11 +148,11 @@ export default {
 
         traversePickingCircle(radius, (x, y) => {
             const idx = (y * 2 * radius + x) * 4;
-            const data = buffer.slice(idx, idx + 4);
+            const data = buffer.slice(idx, idx + 3);
 
             // see PotreeProvider and the construction of unique_id
-            const objId = (data[0] << 8) | data[1];
-            const index = (data[2] << 8) | data[3];
+            const objId = data[0];
+            const index = (data[1] << 16) | data[2];
```

Comment:
You encode each 48-bits `unique_id` value as 3 `uint16`. However, you read the resulting texture with a `uint8` buffer (see [`c3dEngine#renderViewToBuffer`](https://github.com/iTowns/itowns/blob/ee56ec70626213390fa200573552ddb5a54a3b94/src/Renderer/c3DEngine.js#L147))...

I don't think values are preserved here.

Reply:
We can provide a u16 buffer ourselves
```diff
- const buffer = view.mainLoop.gfxEngine.renderViewToBuffer(
-     { camera: view.camera, scene: tileLayer.object3d },
-     {
-         x: viewCoords.x - radius,
-         y: viewCoords.y - radius,
-         width: 1 + radius * 2,
-         height: 1 + radius * 2,
-     });
+ const zone = {
+     x: viewCoords.x - radius,
+     y: viewCoords.y - radius,
+     width: 1 + radius * 2,
+     height: 1 + radius * 2,
+ };
+ zone.buffer = new Uint16Array(4 * zone.x * zone.height);
+ const buffer = view.mainLoop.gfxEngine.renderViewToBuffer(
+     { camera: view.camera, scene: tileLayer.object3d },
+     zone,
+ );
```
But looking into things a bit more I'm not sure how this is gonna interact with the default `fullSizeRenderTarget` used by the `c3DEngine` that gets created with u8 rgba formatting.
I currently don't have an example I could use to check for this issue.